backend/typescript/services/implementations/campService.ts

```diff
@@ -68,6 +66,33 @@ class CampService implements ICampService {
         return [];
       }
 
+      if (year) {
+        // Filter out camps that have camp sessions in the desired year
+        /* eslint-disable no-param-reassign */
+        camps = camps.filter((camp) => {
+          // Go through every camp session in a camp. If it has the desired year, include it in the results.
+          /* eslint-disable no-param-reassign */
+          camp.campSessions = (camp.campSessions as CampSession[]).filter(
+            (campSession) => {
+              return campSession.dates.every((campDate) => {
+                const startCampYearTime = new Date(year, 0, 1).getTime();
+                const endCampYearTime = new Date(year, 11, 31).getTime();
+                return (
+                  campDate.getTime() >= startCampYearTime &&
+                  campDate.getTime() <= endCampYearTime
+                );
+              });
+            },
+          );
+
+          // After filtering all the campSessions that have the wanted year, return all camps that have campSessions in the year or were created in the year
```

Comment:
Edge case: camp query for 2021, should we return a camp created in 2021 with no camp sessions in 2021 (eg. let's say it has 5 camp sessions in 2022)?

This comment + logic says yes, but from the Slack convo I thought we only wanted creation date if no camp sessions exist

Reply:
updated the logic to match Slack convo - will need to remove the check for undefined/null type once our test data all have the createdAt field